src/main/groovy/com/github/jk1/license/filter/LicenseBundleNormalizer.groovy

```diff
@@ -167,72 +173,81 @@ class LicenseBundleNormalizer implements DependencyFilter {
         importedModuleBundle.modules.addAll(normalizedModuleData)
     }
 
-    private Collection<License> normalizePomLicense(License license) {
-        List<NormalizerTransformationRule> rules = [ ]
+    @CompileStatic
+    private Collection<License> normalizePomLicense(License license,
+                                                    String module) {
+        List<NormalizerTransformationRule> rules = transformationRulesFor(module, license.name, license.url)
 
-        rules += findMatchingRulesForName(license.name)
-        rules += findMatchingRulesForUrl(license.url)
         LOGGER.debug("License {} ({}) matches the following rules: [{}]", license.name, license.url, rules.join(","))
 
         if (rules.isEmpty()) return [license]
 
         rules.collect { normalizePomLicense(it, license) }
     }
-    private Collection<ManifestData> normalizeManifestLicense(ManifestData manifest) {
-        List<NormalizerTransformationRule> rules = [ ]
 
-        rules += findMatchingRulesForName(manifest.license)
-        rules += findMatchingRulesForUrl(manifest.license)
+    @CompileStatic
+    private Collection<ManifestData> normalizeManifestLicense(ManifestData manifest,
+                                                              String module) {
+        List<NormalizerTransformationRule> rules = transformationRulesFor(module, null, manifest.license)
+
         LOGGER.debug("License {} ({}) matches the following rules: [{}]", manifest.name, manifest.url, rules.join(","))
 
         if (rules.isEmpty()) return [manifest]
 
         rules.collect { normalizeManifestLicense(it, manifest) }
     }
-    private Collection<LicenseFileDetails> normalizeLicenseFileDetailsLicense(LicenseFileDetails licenseFileDetails) {
+
+    @CompileStatic
+    private Collection<LicenseFileDetails> normalizeLicenseFileDetailsLicense(LicenseFileDetails licenseFileDetails,
+                                                                              String module) {
         if (licenseFileDetails.file == null || licenseFileDetails.file.isEmpty()) return [licenseFileDetails]
 
-        List<NormalizerTransformationRule> rules = [ ]
+        def licenseFileContent = { new File("$config.outputDir/$licenseFileDetails.file").text }.memoize()
 
-        String licenseFileContent = new File("$config.outputDir/$licenseFileDetails.file").text
+        List<NormalizerTransformationRule> rules = normalizerConfig.transformationRules.stream()
```

Comment:
I guess the `stream()` is the normal java8 streaming approach.
Unfortunately, this plugin currently targets java 6. I guess this sitll works as long you execute it with at least java 8, but it will crash when someone uses it with java 6.